crates/weaver_semconv/src/body.rs

```diff
@@ -0,0 +1,243 @@
+            ),
+            "string[]"
+        );
+        assert_eq!(
+            format!(
+                "{}",
+                AttributeType::PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Ints)
+            ),
+            "int[]"
+        );
+        assert_eq!(
+            format!(
+                "{}",
+                AttributeType::PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Doubles)
+            ),
+            "double[]"
+        );
+        assert_eq!(
+            format!(
+                "{}",
+                AttributeType::PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Booleans)
+            ),
+            "boolean[]"
+        );
+        assert_eq!(
+            format!("{}", AttributeType::Template(TemplateTypeSpec::Boolean)),
+            "template[boolean]"
+        );
+        assert_eq!(
+            format!("{}", AttributeType::Template(TemplateTypeSpec::Int)),
+            "template[int]"
+        );
+        assert_eq!(
+            format!("{}", AttributeType::Template(TemplateTypeSpec::Double)),
+            "template[double]"
+        );
+        assert_eq!(
+            format!("{}", AttributeType::Template(TemplateTypeSpec::String)),
+            "template[string]"
+        );
+        assert_eq!(
+            format!("{}", AttributeType::Template(TemplateTypeSpec::Strings)),
+            "template[string[]]"
+        );
+        assert_eq!(
+            format!("{}", AttributeType::Template(TemplateTypeSpec::Ints)),
+            "template[int[]]"
+        );
+        assert_eq!(
+            format!("{}", AttributeType::Template(TemplateTypeSpec::Doubles)),
+            "template[double[]]"
+        );
+        assert_eq!(
+            format!("{}", AttributeType::Template(TemplateTypeSpec::Booleans)),
+            "template[boolean[]]"
+        );
+        assert_eq!(
+            format!(
+                "{}",
+                AttributeType::Enum {
+                    allow_custom_values: true,
+                    members: vec![EnumEntriesSpec {
+                        id: "id".to_owned(),
+                        value: ValueSpec::Int(42),
+                        brief: Some("brief".to_owned()),
+                        note: Some("note".to_owned()),
+                        stability: None,
+                        deprecated: None,
+                    }]
+                }
+            ),
+            "enum {id}"
+        );
+    }
+
+    #[test]
+    fn test_body_field() {
+        let field = BodyFieldSpec {
+            id: "id".to_owned(),
+            r#type: AttributeType::PrimitiveOrArray(PrimitiveOrArrayTypeSpec::Int),
+            brief: Some("brief".to_owned()),
+            examples: Some(Examples::Int(42)),
+            requirement_level: RequirementLevel::Basic(BasicRequirementLevelSpec::Required),
+            note: "note".to_owned(),
+            stability: Some(Stability::Stable),
+            deprecated: Some("deprecated".to_owned()),
+        };
+        assert_eq!(field.id, "id");
```

Comment:
Except for the `is_required` these other asserts feel a little redundant now that the body field is a simple struct...